src/hotspot/share/services/diagnosticArgument.cpp

```diff
@@ -350,3 +354,22 @@ template <> void DCmdArgument<MemorySizeArgument>::init_value(TRAPS) {
 }
 
 template <> void DCmdArgument<MemorySizeArgument>::destroy_value() { }
+
+template <> void DCmdArgument<FileArgument>::destroy_value() {}
+
+template <> void DCmdArgument<FileArgument>::parse_value(const char *str,
+                                            size_t len, TRAPS) {
+  if (str != nullptr) {
+    if (!_value.parse_value(str, len)) {
+      stringStream error_msg;
+      error_msg.print("Invalid file path: %s", str);
```

Comment:
In all likelyhood the only reason Argument::copy_expand... is ever going to fail would be if the expanded string would not fit the buffer in FileArgument. I'd consider a clearer warning here, therefore ("File path invalid or too long: ")